client/src/redux/sagas/pullsTestmanSaga.js

```diff
@@ -0,0 +1,27 @@
+import { takeEvery, call, select, put } from 'redux-saga/effects';
+import { PULLS } from '../constants';
+import { fetchTests } from '../api';
+import { setTestman, setTestmanError } from '../actions';
+
+function* handleTestmanLoad() {
+  try {
+    const pulls = yield select(state => state.pulls);
+    const page = 1;
+    const shas = pulls.map(pull => pull.merge_commit_sha);
+    const testResults = yield call(fetchTests, shas[9], shas[0], page);
+    const testByPulls = {};
+    for (let { merge_commit_sha } of pulls) {
+      const testData = testResults.filter(test =>
+        merge_commit_sha.includes(test.revision._text)
```

Comment:
Just to be sure: you use `includes` here because the revision from testman contains only short hash?

Reply:
yes, revision from testman contains only short hashes. till now I haven't seen full revison in any of the request

Reply:
https://github.com/reactos/web/blob/3e783c2520f6ec8f4451c9d51583209aee69309c/www/www.reactos.org/testman/ajax-search.php#L121    here getShortHash is sent in output so it's ok to use `includes`

Reply:
I think `String.prototype.startsWith()` would be better here